btf/btf.go

```diff
@@ -211,8 +227,17 @@ func loadSpecFromELF(file *internal.SafeELFFile) (*Spec, error) {
 	return spec, nil
 }
 
-func loadRawSpec(btf io.ReaderAt, bo binary.ByteOrder,
-	baseTypes types, baseStrings *stringTable) (*Spec, error) {
+func loadRawSpec(btf io.ReaderAt, bo binary.ByteOrder, base *Spec) (*Spec, error) {
+	var baseStrings *stringTable
+	var baseTypes types
+	if base != nil {
+		baseStrings = base.strings
+		baseTypes = base.types
+
+		if baseStrings == nil {
```

Comment:
A valid string table always has at least the empty string, so the single anonymous type is not a problem.

We call this from two places, LoadSplitSpec... and Handle.Spec. I wanted to avoid duplicating the logic, but maybe I went overboard. What if I change `loadRawSpec` to take `baseStrings, baseTypes`?

Reply:
Moved!